--- www/lib/framework/basepage.php	Sat Apr 20 20:56:07 2013
+++ www/lib/framework/basepage.php	Mon Apr 22 09:54:11 2013
@@ -66,6 +66,8 @@
 		$users = new Users();
 		if ($users->isLoggedIn())
 		{
+            // default content for logged in users.
+            $this->page = (isset($_GET['page'])) ? $_GET['page'] : ltrim($this->site->home_link, '/');
 			$this->userdata = $users->getById($users->currentUserId());
 			
 			//
@@ -106,6 +108,9 @@
 		}
 		else
 		{
+            // default content for unlogged in users.
+            $this->page = (isset($_GET['page'])) ? $_GET['page'] : 'content';
+
 			$this->smarty->assign('isadmin',"false");	
 			$this->smarty->assign('loggedin',"false");	
 			$this->floodCheck(false, "");
--- www/lib/releasecomments.php	Mon Apr 15 22:44:07 2013
+++ www/lib/releasecomments.php	Thu Apr 25 09:05:24 2013
@@ -22,7 +22,7 @@
 	public function getCommentsByGid($gid)
 	{			
 		$db = new DB();
-		return $db->query(sprintf("SELECT rc.id, text, createddate, sourceid, CASE WHEN sourceid = 0 THEN (SELECT username FROM users WHERE id = userid) ELSE username END AS username FROM releasecomment rc WHERE isvisible = 1  AND gid = %s AND (userID IN (SELECT id FROM users) OR rc.username IS NOT NULL) ORDER BY createddate DESC LIMIT 100", $db->escapeString($gid)));				
+		return $db->query(sprintf("SELECT rc.id, text, createddate, sourceid, CASE WHEN sourceid = 0 THEN (SELECT username FROM users WHERE id = userid) ELSE username END AS username, CASE WHEN sourceid = 0 THEN (SELECT role FROM users WHERE id = userid) ELSE '-1' END AS role, CASE WHEN sourceid =0 THEN (SELECT r.name AS rolename FROM users AS u LEFT JOIN userroles AS r ON r.ID = u.role WHERE u.id = userid) ELSE (SELECT description AS rolename FROM spotnabsources WHERE ID = sourceid) END AS rolename FROM releasecomment rc WHERE isvisible = 1  AND gid = %s AND (userID IN (SELECT id FROM users) OR rc.username IS NOT NULL) ORDER BY createddate DESC LIMIT 100", $db->escapeString($gid)));				
 	}
 
 	/**
@@ -31,7 +31,7 @@
 	public function getCommentsByGuid($guid)
 	{			
 		$db = new DB();
-		return $db->query(sprintf("SELECT rc.id, text, createddate, sourceid, CASE WHEN sourceid = 0 THEN (SELECT username FROM users WHERE id = userid) ELSE username END AS username FROM releasecomment rc LEFT JOIN releases r ON r.gid = rc.gid WHERE isvisible = 1 AND guid = %s AND (userID IN (SELECT id FROM users) OR rc.username IS NOT NULL) ORDER BY createddate DESC LIMIT 100", $db->escapeString($guid)));	
+		return $db->query(sprintf("SELECT rc.id, text, createddate, sourceid, CASE WHEN sourceid = 0 THEN (SELECT username FROM users WHERE id = userid) ELSE username END AS username, CASE WHEN sourceid = 0 THEN (SELECT role FROM users WHERE id = userid) ELSE '-1' END AS role, CASE WHEN sourceid =0 THEN (SELECT r.name AS rolename FROM users AS u LEFT JOIN userroles AS r ON r.ID = u.role WHERE u.id = userid) ELSE (SELECT description AS rolename FROM spotnabsources WHERE ID = sourceid) END AS rolename FROM releasecomment rc LEFT JOIN releases r ON r.gid = rc.gid WHERE isvisible = 1 AND guid = %s AND (userID IN (SELECT id FROM users) OR rc.username IS NOT NULL) ORDER BY createddate DESC LIMIT 100", $db->escapeString($guid)));	
 	}
 
 	/**
--- www/pages/browse.php	Wed Apr 24 15:58:53 2013
+++ www/pages/browse.php	Wed Apr 24 14:50:46 2013
@@ -1,8 +1,10 @@
 <?php
 require_once(WWW_DIR."/lib/releases.php");
+require_once(WWW_DIR."/lib/releaseextra.php");
 require_once(WWW_DIR."/lib/category.php");
 
 $releases = new Releases;
+$re = new ReleaseExtra;
 
 if (!$users->isLoggedIn())
         $page->show403();
@@ -27,6 +29,19 @@
 
 $results = array();
 $results = $releases->getBrowseRange($catarray, $offset, ITEMS_PER_PAGE, $orderby, -1, $page->userdata["categoryexclusions"], $grp);
+
+foreach($results as $key => $value)
+{
+    $results[$key]['reAudio'] = $re->getAudio($value['ID']);
+    $results[$key]['reVideo'] = $re->getVideo($value['ID']);
+    if($results[$key]['reVideo'] != "")
+    {
+        $width = $results[$key]['reVideo']['videowidth'];
+        $height = $results[$key]['reVideo']['videoheight'];
+        $results[$key]['reRes'] = $re->determineVideoResolution($width, $height);
+    }
+}
+
 
 $page->smarty->assign('pagertotalitems',$browsecount);
 $page->smarty->assign('pageroffset',$offset);
--- www/pages/search.php	Mon Apr 15 22:46:07 2013
+++ www/pages/search.php	Wed Apr 24 14:51:22 2013
@@ -1,9 +1,11 @@
 <?php
 require_once(WWW_DIR."/lib/releases.php");
+require_once(WWW_DIR."/lib/releaseextra.php");
 require_once(WWW_DIR."/lib/groups.php");
 require_once(WWW_DIR."/lib/category.php");
 
 $releases = new Releases;
+$re = new ReleaseExtra;
 $grp = new Groups;
 $c = new Category;
 
@@ -47,6 +49,18 @@
 		$page->smarty->assign('search', $searchStr);
 
 		$results = $releases->search($searchStr, $categoryId, $offset, ITEMS_PER_PAGE, $orderby, -1, $page->userdata["categoryexclusions"]);
+        
+        foreach($results as $key => $value)
+        {
+            $results[$key]['reAudio'] = $re->getAudio($value['ID']);
+            $results[$key]['reVideo'] = $re->getVideo($value['ID']);
+            if($results[$key]['reVideo'] != "")
+            {
+                $width = $results[$key]['reVideo']['videowidth'];
+                $height = $results[$key]['reVideo']['videoheight'];
+                $results[$key]['reRes'] = $re->determineVideoResolution($width, $height);
+            }
+        }
 
 	}
 	else
